### ISSUE TITLE
Accessibility: Table header missing for non-empty data cell in job_status_table

### ISSUE THREAD

**Issue by folkwest** [OSC/ondemand#5185]
**Description**
An accessibility issue was identified in the #job_status_table component. According to WCAG guidelines, each non-empty data cell in a table larger than 3×3 must have one or more associated table headers. Currently, some non-empty data cells in the job status table do not have proper table header associations, which creates barriers for screen reader users and other assistive technologies.

**Affected Element**

<img width="918" height="212" alt="Image" src="https://github.com/user-attachments/assets/cc128615-9810-43d3-a072-fef51eeb1250" />

`<table id="job_status_table" class="table datatable table-hover dataTable no-footer" cellspacing="0" width="100%" aria-describedby="job_status_table_info" style="width: 100%;">`

**Related Nodes**
The following element was flagged as part of the issue:
`<td><button class="details-control fa fa-chevron-right btn btn-default" aria-expanded="false" aria-label="Toggle visibility of job details for job X on Klone"></button></td>`

**Suggested Fix**
Ensure all `<td>` elements have corresponding `<th>` parent elements in the table header row
For control buttons in the first column (like the expand/collapse button), consider:

- Adding a proper header cell (`<th>`) for that column
- Using scope="col" on the header element

Example corrected structure:
```
<thead>
  <tr>
    <th scope="col">Actions</th>
    <th scope="col">Job Name</th>
    <!-- other headers -->
  </tr>
</thead>
```

**Comment by johrstrom** [comment]
Thanks. Yea we're working through that whole table now, hopefully it'll be a lot better.

**Comment by folkwest** [comment]
Sounds great! Wanted to flag just in case. Thank you for the work that you're doing!

**Comment by moffatsadeghi** [comment]
Thank you so much for opening these issues! We really appreciate it

**Comment by johrstrom** [comment]
This may be fixed in #5181, but I'll leave it open to confirm the header works as expected.

**Comment by moffatsadeghi** [comment]
Just need to confirm that the header works as expected.  Will be assigning to @johrstrom to do so :)

**Comment by johrstrom** [comment]
Yes this has been fixed in 4.1. #5181 did in fact add a column header.
